src/WindowManager.vala

```diff
@@ -2042,10 +2042,14 @@ namespace Gala {
                 return false;
 
             var modal_proxy = modal_stack.peek_head ();
+            if (modal_proxy != null) {
```

Comment:
Nitpick, so feel free to ignore it, but I think this could be easier to follow returning earlier:

```
var modal_proxy = modal_stack.peek_head ();
if (!modal_proxy) {
    return false;
}

var filter = modal_proxy.get_keybinding_filter ();
if (!filter) {
    return false;
}

return filter (binding);
```

Reply:
I dislike having more than 2 `return` s in a function if it can be helped but I can see less nesting could be clearer.

Reply:
Fair enough, both approaches are fine. Resolving this conversation 👍